.github/workflows/ci.yml

```diff
@@ -19,32 +17,40 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ "ubuntu-latest", "macos-latest"]
-        python-version: [ "3.7", "3.8", "3.9", "3.10" ]
+        os: [ "ubuntu-latest", "macos-latest", "macos-14" ]
+        python-version: [ "3.9", "3.10", "3.11" ]
 
     steps:
       - name: Cancel previous runs
-        uses: styfle/cancel-workflow-action@0.9.1
+        uses: styfle/cancel-workflow-action@0.11.0
         with:
           access_token: ${{ github.token }}
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           token: ${{ github.token }}
-      - name: Conda setup
+      - name: Conda setup (Linux)
         uses: conda-incubator/setup-miniconda@v2
+        if: matrix.os == 'ubuntu-latest' 
         with:
-          activate-environment: wrf_python_build
           python-version: ${{ matrix.python-version }}
-          channels: conda-forge, ncar
-      - name: Conda install (Darwin)
+          channels: conda-forge
+          environment-file: build_envs/Linux.yml
+      - name: Conda setup (macOS)
+        uses: conda-incubator/setup-miniconda@v2
         if: matrix.os == 'macos-latest'
-        run: |
-          conda env update --file build_envs/Darwin.yml --prune
-      - name: Conda install (Linux)
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          conda env update --file build_envs/Linux.yml --prune
+        with:
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+          environment-file: build_envs/Darwin.yml
+      - name: Conda setup (macOS M1)
+        uses: conda-incubator/setup-miniconda@v2
+        if: matrix.os == 'macos-14'
+        with:
+          installer-url: https://github.com/conda-forge/miniforge/releases/download/23.11.0-0/Mambaforge-23.11.0-0-MacOSX-arm64.sh
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+          environment-file: build_envs/environment.yml
```

Comment:
That's nice to see having just a `compilers` dependency in this env file could take care of those M1 compilers needed. Though, I have a few comments for which I am curious about your thoughts:

- It's also helpful to see those actually needed compilers being explicitly listed in each platform-specific env file. Could we still also create a `MacOS_M1.yml` file that'd have only the M1-specific compilers listed?
- Having this `environment.yml` as a more universal file still makes a lot of sense to me. 
   - Should we see if it could work for all of the platform-specific action runs? If so, should we cleanup this CI file siginificantly by just using that for all the platforms?  
      - If this'd be the case, I'd suggest we document the purpose for this universal one and other platform-specific ones well for the user and developers.

Reply:
Yeah, that (a single environment file) was where I was heading with this.  Scipy and Numpy seem to do this rather than specifying compilers and env files for each OS.  The `compilers` dependency apparently just grabs the ones used by conda-forge for your OS.

Rather than overhauling everything, I set this up as a demo of sorts.  Though I think it probably makes sense to just make the change for all platforms.  Having both the universal `environment.yml` that we use here and the platform specific ones in case they are still needed for dev seems like a good plan for now at least.  

From a maintenance perspective, I am a little concerned that the platform specific env files might go stale (I think some may already fall in that category).  If those aren't regularly used somewhere else and/or clearly needed, I the single `environment.yml` file might be the way to go.  That said, I'm a bit new this and don't know all the history here, so very curious what others think.

For now I'll go ahead and see if I can get the universal `environment.yml` file working for all of these runs and add the M1 specific env file.

Reply:
Yes, maintaining a universal `environment.yml` file for all platforms seems to be the most cost-effective way. Please give it a try, and if it works for all of the platforms, why don't we archive those platform-specific env files in such a way:

- Rather than removing them at all, put them into a new folder (for now), so the `build_envs` would look like as follows:
   - /build_envs
      - `environment.yml`
      - /platform-specific
        - README
        - `Linux.yml`
        - `MacOS_x64.yml`
        - `MacOS_arm64.yml`
        - `Win64.yml` 

where  README would explain that we use and maintain only `environment.yml` and provide others for the sole purpose of reference but NOT maintain them
-  Get rid of their corresponding 'conda setup"s in CI to clean it up to use only `environment.yml`
- In the future, maybe we'd get rid of those archived `.yml` files at all